CRM/Core/DAO.php

```diff
@@ -636,6 +642,10 @@ public function copyValues(&$params) {
         if ($pValue === '') {
           $this->$dbName = 'null';
         }
+        elseif ($serializeArrays && is_array($pValue) && !empty($value['serialize'])) {
+          $this->$dbName = CRM_Core_DAO::serializeField($pValue, $value['serialize']);
+          $allNull = FALSE;
+        }
```

Comment:
I appreciate your caution, but to my eye this looks like it ought to be safe, since the function `copyValues` is called before writing an object and those values *do* need to be serialized.
There may be a few cases where `copyValues` is called and then a value passed in as an array is immediately overwritten, but this change wouldn't break that type of hack (and said hack can be removed once this is merged).

Reply:
What I worry about is this

$dao->copyValues
if ($dao->serialisedField) {
  $dao->serialisedField = serialize($dao->serialisedField);
}
$dao->save()

Reply:
Understood. I don't know if that type of horrible code exists but until we get a chance to crawl through the codebase and verify it's not there, I'll merge this cautious version for now.

Reply:
Hold the phone, the above snippet would work just fine. Check the implementation of `DAO->serializeField()` and how it handles non-array values :)
@eileenmcnaughton I think we don't need the extra caution in light of that.

Reply:
@colemanw my worry is if array values are handled twice - we can work to remove it again.

Reply:
after adding more testing in more places & more grepping